### PR TITLE
controller: add unit test to prove debug and metrics are served

### PR DESCRIPTION
**Pull request by mfojtik**

### CODE REVIEW

File: pkg/controller/controllercmd/builder.go [hunk 1]
```diff
@@ -251,12 +278,13 @@ func (b *ControllerBuilder) Run(ctx context.Context, config *unstructured.Unstru
...
 			return err
 		}
-		serverConfig.Authorization.Authorizer = union.New(
-			// prefix the authorizer with the permissions for metrics scraping which are well known.
-			// openshift RBAC policy will always allow this user to read metrics.
-			hardcodedauthorizer.NewHardCodedMetricsAuthorizer(),
-			serverConfig.Authorization.Authorizer,
-		)
+
+		if b.authorizer != nil {
```
Comment (deads2k) [inline pkg/controller/controllercmd/builder.go:282]: so if it is nil, we get no authorizer?  that seems bad.

File: pkg/controller/controllercmd/builder.go [hunk 2]
```diff
@@ -190,35 +201,51 @@ func (b *ControllerBuilder) WithEventRecorderOptions(options record.CorrelatorOp
 
 // Run starts your controller for you.  It uses leader election if you asked, otherwise it directly calls you
 func (b *ControllerBuilder) Run(ctx context.Context, config *unstructured.Unstructured) error {
-	clientConfig, err := b.getClientConfig()
-	if err != nil {
-		return err
+	var clientConfig *rest.Config
+
+	if b.kubeAPIServerConfigFile != nil {
```
Comment (deads2k) [inline pkg/controller/controllercmd/builder.go:206]: this change appears to have broken the automatic in-cluster configuration getting of a client.

Comment (deads2k) [inline pkg/controller/controllercmd/builder.go:206]: this seems like something to include in `b.getClientConfig`

Comment (deads2k) [reply pkg/controller/controllercmd/builder.go:206]: > this seems like something to include in `b.getClientConfig`

and in fact it is.

File: pkg/controller/controllercmd/builder.go [hunk 3]
```diff
@@ -190,35 +201,51 @@ func (b *ControllerBuilder) WithEventRecorderOptions(options record.CorrelatorOp
...
 	}
-	controllerRef, err := events.GetControllerReferenceForCurrentPod(kubeClient, namespace, nil)
-	if err != nil {
-		klog.Warningf("unable to get owner reference (falling back to namespace): %v", err)
+
+	var (
+		kubeClient    kubernetes.Interface
+		controllerRef *corev1.ObjectReference
+	)
+	if clientConfig != nil {
```
Comment (deads2k) [inline pkg/controller/controllercmd/builder.go:227]: if the client-config is nil when you create it, just fail then.

File: pkg/controller/controllercmd/cmd.go [hunk 4]
```diff
@@ -90,9 +90,6 @@ func (c *ControllerCommandConfig) NewCommandWithContext(ctx context.Context) *co
 
 			defer logs.FlushLogs()
 			defer serviceability.BehaviorOnPanic(os.Getenv("OPENSHIFT_ON_PANIC"), c.version)()
-			defer serviceability.Profile(os.Getenv("OPENSHIFT_PROFILE")).Stop()
-
-			serviceability.StartProfiler()
```
Comment (deads2k) [inline pkg/controller/controllercmd/cmd.go:95]: why did you kill our profiler?

**Comment by deads2k** [comment]
Seems like what you really want to do is create an e2e test that checks every operator pod to see if it serves debug endpoints.

File: pkg/controller/controllercmd/builder.go (hunk 1 above)
Comment (mfojtik) [reply pkg/controller/controllercmd/builder.go:282]: this is never `nil` except unit test or when somebody generate operator without WithServer (which in that case it does not matter).

File: pkg/controller/controllercmd/cmd.go (hunk 4 above)
Comment (mfojtik) [reply pkg/controller/controllercmd/cmd.go:95]: there is already profiler in apiserver :-) so in fact we running 2 profilers today

**Pull request closed without merging**